### PR TITLE
feat: enabling the new condition

### PR DESCRIPTION
**Pull request by zhiying-lin**
### Description of your changes

Fixes #

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

**Review by ryanzhang-oss — APPROVED** [review]

**Review by zhiying-lin — COMMENTED** [review]
LGTM

File: pkg/controllers/rollout/controller.go [hunk 1]
```diff
@@ -380,14 +380,15 @@ func (r *Reconciler) pickBindingsToRoll(ctx context.Context, allBindings []*flee
 				klog.V(3).InfoS("Found a ready bound binding", "clusterResourcePlacement", crpKObj, "binding", bindingKObj)
 				readyBindings = append(readyBindings, binding)
 			}
+			// check if the binding is failed or still on going
 			appliedCondition := binding.GetCondition(string(fleetv1beta1.ResourceBindingApplied))
-			if condition.IsConditionStatusFalse(appliedCondition, binding.Generation) {
-				klog.V(3).InfoS("Found a failed to apply bound binding", "clusterResourcePlacement", crpKObj, "binding", bindingKObj)
+			availableCondition := binding.GetCondition(string(fleetv1beta1.ResourceBindingAvailable))
```
Comment (zhiying-lin) [inline pkg/controllers/rollout/controller.go:385]: need to fix one more place, let me create a commit

File: apis/placement/v1beta1/binding_types.go [hunk 2]
```diff
@@ -123,14 +123,6 @@ const (
 	// - "Unknown" means it is unknown.
 	ResourceBindingOverridden ResourceBindingConditionType = "Overridden"
 
-	// ResourceBindingBound indicates the bound condition of the given resources.
-	// Its condition status can be one of the following:
-	// - "True" means the corresponding work CR is created in the target cluster's namespace.
-	// - "False" means the corresponding work CR is not created yet.
-	// - "Unknown" means it is unknown.
-	// TODO, will be replaced by "WorkSynchronized"
-	ResourceBindingBound ResourceBindingConditionType = "Bound"
```
Comment (michaelawyu) [inline apis/placement/v1beta1/binding_types.go:132]: Hi Zhiying! Could you do the removal in the v1 package as well? I know it wouldn't affect the CRD, but a little bit consistency would still be better I think

File: test/e2e/setup.sh [hunk 3]
```diff
@@ -117,7 +117,6 @@ helm install hub-agent ../../charts/hub-agent/ \
     --set image.pullPolicy=Never \
     --set image.repository=$REGISTRY/$HUB_AGENT_IMAGE \
     --set image.tag=$TAG \
-    --set logVerbosity=2 \
```
Comment (michaelawyu) [inline test/e2e/setup.sh:120]: How about we change it to 5 instead?

**Review by ryanzhang-oss — APPROVED** [review]
rollout needs more integration tests

**Pull request merged**